src/quill.mention.js

```diff
@@ -160,11 +163,28 @@ class Mention {
     return this.allowedChars.test(s);
   }
 
-  setMentionListPosition(startIndex) {
+  setMentionListPosition(atSignIndex) {
     const containerPos = this.quill.container.getBoundingClientRect();
-    const indexPos = this.quill.getBounds(startIndex);
-    this.mentionContainer.style.top = `${window.scrollY + containerPos.top + indexPos.bottom}px`;
-    this.mentionContainer.style.left = `${window.scrollX + containerPos.left + indexPos.left}px`;
+    const atSignBounds = this.quill.getBounds(atSignIndex);
+
+    if ((atSignBounds.left + 230) > this.quill.container.offsetWidth) {
+      this.mentionContainer.style.left = "auto";
+      this.mentionContainer.style.right = 0;
+    } else {
+      this.mentionContainer.style.left = atSignBounds.left + "px";
+    }
+
+    let windowHeight = window.innerHeight;
+    let editorPos = this.quill.container.getBoundingClientRect().top;
+
+    if (editorPos > windowHeight / 2) {
+      this.mentionContainer.style.top = "auto";
+      this.mentionContainer.style.bottom = atSignBounds.top + atSignBounds.height + 15 + "px";
```

Comment:
15 is the font size? Should be something like atSignBounds.bottom instead

Reply:
Honestly, I'm not sure. I copied that code, but don't fully understand it. I'll look at it some more on Monday.